crates/transpiler/src/target/mod.rs

```diff
@@ -891,6 +895,381 @@ impl Target {
+    }
+
+    /// Sets a list of the characteristics of each qubit on the [Target] device.
+    ///
+    /// **Warning:** Should only be used during the construction process of [Target],
+    /// after calling [Target::new] or any other construction methods.
+    ///
+    /// # Arguments
+    ///
+    /// * `qubit_properties` - A list of python `QubitProperties` objects defining
+    ///        the characteristics of each qubit on the target device. If specified
+    ///        the length of this list must match the number of qubits in the target,
+    ///        where the index in the list matches the qubit number the properties
+    ///        are defined for. If some qubits don't have properties available you
+    ///        can set that entry to ``None``.
+    ///
+    /// # Returns
+    ///
+    /// Initialized [Target] with a defined ``qubit_properties`` attribute.
+    ///
+    /// # Panics
+    ///
+    /// This function panics if there was a specified [Target::num_qubits] attribute and the
+    /// the length of the provided list is not equal to [Target::num_qubits].
+    ///
+    /// /// # Examples
+    ///
+    /// ```rust
+    /// use qiskit_accelerate::target_transpiler::Target;
+    ///
+    /// let mut target = Target::new().with_qubit_properties(vec![]);
+    /// assert_eq!(target.num_qubits, Some(0));
+    /// ```
+    pub fn with_qubit_properties(self, qubit_properties: Vec<PyObject>) -> Self {
+        self.try_with_qubit_properties(qubit_properties)
+            .unwrap_or_else(|e| panic!("{}", e.to_string()))
+    }
+
+    /// Sets a list of the characteristics of each qubit on the [Target] device.
+    ///
+    /// **Warning:** Should only be used during the construction process of [Target],
+    /// after calling [Target::new] or any other construction methods.
+    ///
+    /// # Arguments
+    ///
+    /// * `qubit_properties` - A list of python `QubitProperties` objects defining
+    ///        the characteristics of each qubit on the target device. If specified
+    ///        the length of this list must match the number of qubits in the target,
+    ///        where the index in the list matches the qubit number the properties
+    ///        are defined for. If some qubits don't have properties available you
+    ///        can set that entry to ``None``.
+    ///
+    /// # Returns
+    ///
+    /// * `Ok`: (if the`qubit_properties` were successfully set.) [Target]
+    /// * `Err`: (if there was a specified [Target::num_qubits] attribute and the
+    ///     lengths did not match) [TargetError].
+    ///
+    /// /// # Examples
+    ///
+    /// ```rust
+    /// use qiskit_accelerate::target_transpiler::Target;
+    ///
+    /// let mut target = Target::new().try_with_qubit_properties(vec![]);
+    /// assert!(target.is_ok());
+    /// // This should change the number of qubits to the length of the list.
+    /// assert_eq!(target.num_qubits, Some(0));
+    /// ```
+    pub fn try_with_qubit_properties(
```

Comment:
We definitely shouldn't have `qubit_properties` as `pub` if we have this method. I did mention this before, but seeing it explicitly here we definitely need to remove the `pub` since it's unsound to have it mutable directly.

Reply:
Made private in 73228f5